engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java

```diff
@@ -42,4 +43,59 @@ public void wrongFunctionSignature() throws Exception {
       assertTrue("It is a syntax error exception", e.isSyntaxError());
     }
   }
+
+  @Test
+  public void wrongAscribedTypeSignature() throws Exception {
+    final URI uri = new URI("memory://neg.enso");
+    final Source src = Source.newBuilder("enso", """
+    neg (a : Xyz) = 0 - a
+    """, uri.getHost())
+            .uri(uri)
+            .buildLiteral();
+
+    try {
+      var module = ctx.eval(src);
+      var neg = module.invokeMember("eval_expression", "neg");
+      fail("Expecting an exception from compilation, not: " + neg);
+    } catch (PolyglotException e) {
+      assertTrue("It is a syntax error exception", e.isSyntaxError());
+    }
+  }
+
+  @Test
+  public void wrongAscribedInConstructor() throws Exception {
+    final URI uri = new URI("memory://constructor.enso");
+    final Source src = Source.newBuilder("enso", """
+    type Neg
+      Val (a : Xyz)
+
+    neg = Neg.Val 10
+    """, uri.getHost())
+            .uri(uri)
+            .buildLiteral();
+
+    try {
+      var module = ctx.eval(src);
+      var neg = module.invokeMember("eval_expression", "neg");
+      fail("Expecting an exception from compilation, not: " + neg);
+    } catch (PolyglotException e) {
+      assertTrue("It is a syntax error exception", e.isSyntaxError());
+    }
+  }
+
+  @Test
+  public void ascribedWithAParameter() throws Exception {
+    final URI uri = new URI("memory://constructor.enso");
+    final Source src = Source.newBuilder("enso", """
+    type Maybe a
+        Nothing
+        Some unwrap:a
```

Comment:
This test mimics failure in `org.enso.interpreter.test.semantic.ImportsTest` which has this type in `engine/runtime/src/test/resources/Test_Type_Imports/src/Type_Def.enso`